egs/swbd/s5c/local/nnet3/multi_condition/run_ivector_common.sh

```diff
@@ -0,0 +1,148 @@
+#!/bin/bash 
+#set -e
+# this script is based on local/nnet3/run_ivector_common.sh
+# but it operates on corrupted training/dev/test data sets
+
+. cmd.sh
+
+stage=1
+foreground_snrs="20:10:15:5:0"
+background_snrs="20:10:15:5:0"
+num_data_reps=1
+clean_data_dir=train_nodup_sp
+iv_dir=exp/nnet3_rvb
+
+set -e
+. cmd.sh
+. ./path.sh
+. ./utils/parse_options.sh
+
+mkdir -p $iv_dir
+train_set=${clean_data_dir}_rvb${num_data_reps}
+
+if [ $stage -le 1 ]; then
+  if [ ! -d "RIRS_NOISES" ]; then
+    # Download the package that includes the real RIRs, simulated RIRs, isotropic noises and point-source noises
+    wget --no-check-certificate http://www.openslr.org/resources/28/rirs_noises.zip
+    unzip rirs_noises.zip
+  fi
+
+  # corrupt the data to generate reverberated data 
+  python steps/data/reverberate_data_dir.py \
+    --prefix "rev" \
+    --rir-set-parameters "0.25, RIRS_NOISES/simulated_rirs/smallroom/rir_list" \
+    --rir-set-parameters "0.25, RIRS_NOISES/simulated_rirs/mediumroom/rir_list" \
+    --rir-set-parameters "0.25, RIRS_NOISES/simulated_rirs/largeroom/rir_list" \
+    --rir-set-parameters "0.25, RIRS_NOISES/real_rirs_isotropic_noises/rir_list" \
+    --foreground-snrs $foreground_snrs \
+    --background-snrs $background_snrs \
+    --speech-rvb-probability 1 \
+    --pointsource-noise-addition-probability 1 \
+    --isotropic-noise-addition-probability 1 \
+    --num-replications $num_data_reps \
+    --max-noises-per-minute 1 \
+    --source-sampling-rate 8000 \
+    data/${clean_data_dir} data/${train_set}
```

Comment:
Some more comments after looking at the script in more detail:
If we want to use this script as a starting point for how to do data reverberation it's going to be quite hard.

For a start, it has a very messy dependency on the regular run_ivector_common.sh, because it assumes that the train_nodup_sp directory already exists.   There are scenarios where we'd want to run the reverberation stuff from scratch without running the regular run_ivector_common.sh.  It would be better to have this script do both the speed perturbing and the RIR stuff; to avoid overwriting features that might have been already written by run_ivector_common.sh, you could put that step first so the --stage parameter can be used to avoid redoing stuff, and have the script die if the feats.scp exists in the _sp directory (and would be overwritten if the script were to run).
To keep things simple, you can just make the speed-perturbing non-optional.  I don't think anyone will want to do this without speed perturbing; if you want to do so for your own experiments you can do it locally.

Also, this script seems to have a very detailed dependency on the Switchboard setup, because of the way you prepare the data subset for the lda_mllt stage, and the way you re-use the alignments.
The LDA+MLLT transform is extremely non-critical, because it only affects the diagonal GMM which is just used to initialize the full GMM and to pre-prune when getting GMM posteriors.  And the number of parameters is tiny.  You could use a very small data subset for that, and it would be best to eliminate all of the detailed dependency on the Switchboard setup.  What I think would make more sense is, after you create the _mix data, to select a fairly small subset of it (with a num-utts that can be specified on the command line, like 20k), and to dump parallel features with regular MFCC and _hires, and then use a supplied model [which should be defined at the top of the script, and not specified in the body of the script], to get the alignments so you can train a system on the _hires features.

Also, with regard to the way you're mixing it with the un-perturbed data (_mix)...
is there a reason why it's better to mix it manually like that, instead of specifying (say) a number less than 1 for the following:
  --speech-rvb-probability 1 \
    --pointsource-noise-addition-probability 1 \
    --isotropic-noise-addition-probability 1 \
and maybe using 2 replications instead of 1?  It seems to me that that would be more elegant and simpler, if the results were the same.